lms/djangoapps/certificates/api.py

```diff
@@ -484,49 +484,46 @@ def get_active_web_certificate(course, is_preview_mode=None):
     return None
 
 
-def get_certificate_template(course_key, mode, language):  # pylint: disable=unused-argument
+def get_certificate_template(course_key, mode, language):
     """
     Retrieves the custom certificate template based on course_key, mode, and language.
     """
-    org_id, template = None, None
     # fetch organization of the course
-    course_organization = get_course_organizations(course_key)
-    if course_organization:
-        org_id = course_organization[0]['id']
-
-    if org_id and mode:
-        template = CertificateTemplate.objects.filter(
-            organization_id=org_id,
-            course_key=course_key,
-            mode=mode,
-            is_active=True
-        )
-    # if don't template find by org and mode
-    if not template and org_id and mode:
-        template = CertificateTemplate.objects.filter(
-            organization_id=org_id,
-            course_key=CourseKeyField.Empty,
-            mode=mode,
-            is_active=True
-        )
-    # if don't template find by only org
-    if not template and org_id:
-        template = CertificateTemplate.objects.filter(
-            organization_id=org_id,
-            course_key=CourseKeyField.Empty,
-            mode=None,
-            is_active=True
-        )
-    # if we still don't template find by only course mode
-    if not template and mode:
-        template = CertificateTemplate.objects.filter(
-            organization_id=None,
-            course_key=CourseKeyField.Empty,
-            mode=mode,
-            is_active=True
-        )
+    org_id = get_course_organization_id(course_key)
+
+    # prepare template sets.
+    active_templates = CertificateTemplate.objects.filter(is_active=True)
+    org_templates = active_templates.filter(organization_id=org_id)
+    mode_templates = active_templates.filter(mode=mode)
+    org_and_mode_templates = org_templates & mode_templates
```

Comment:
I'm not sure, a couple sites implied that it would (since I think under the hood it just applies both filters), for now I've just switched to to a simple filter since I think its as clear